narwhal/primary/src/header_waiter.rs

```diff
@@ -330,9 +339,12 @@ impl HeaderWaiter {
                         },
                         Ok(header) => {
                             self.cleanup_pending_requests(&header);
-
                             // Ok to drop the header if core is overloaded.
-                            let _ = self.tx_headers_loopback.try_send(header);
+                            if let Err(e) = self.tx_headers_loopback.try_send(header.clone()) {
+                                warn!("Failed to loop back ready header {header} {e}");
```

Comment:
Similar, could you please print the debug version of `header` which is reacher? `{:?}`

Reply:
Same above. This already includes the round number and digest which is what we have been relying on.